src/Resolution.ts

```diff
@@ -530,29 +434,37 @@ export default class Resolution {
 
   private getDnsRecordKeys(types: DnsRecordType[]): string[] {
     const records = ['dns.ttl'];
-    types.forEach(type => {
+    types.forEach((type) => {
       records.push(`dns.${type}`);
       records.push(`dns.${type}.ttl`);
     });
     return records;
   }
 
-  private async getPreferableNewRecord(domain: string, newRecord: string, oldRecord: string): Promise<string> {
-    const records = await this.records(domain, [newRecord, oldRecord]) as Record<string, string>;
-    return NamingService.ensureRecordPresence(domain, newRecord, records[newRecord] || records[oldRecord]);
+  private async getPreferableNewRecord(
+    domain: string,
+    newRecord: string,
+    oldRecord: string,
+  ): Promise<string> {
+    const records = (await this.records(domain, [
+      newRecord,
+      oldRecord,
+    ]));
+    return ensureRecordPresence(
+      domain,
+      newRecord,
+      records[newRecord] || records[oldRecord],
+    );
   }
 
   private getNamingMethod(domain: string): NamingService | undefined {
-    return this.getResolutionMethods().find(method =>
+    return this.getResolutionMethods().find((method) =>
       method.isSupportedDomain(domain),
     );
   }
 
   private getResolutionMethods(): NamingService[] {
-    return (this.blockchain
-      ? ([this.ens, this.zns, this.cns] as NamingService[])
-      : ([this.api] as NamingService[])
-    ).filter(v => v);
+    return ([this.ens, this.zns, this.cns] as NamingService[]).filter((v) => v);
```

Comment:
@JohnnyJumper This is bad. Why we can't use hash map and just detect naming service type in advance? Then just get naming service by key.

Data type example:
```
private readonly namingServices: Record<NamingServiceName, NamingService>;
```

```